### PR TITLE
Fixes #24483 - Provide raw output API endpoint

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by adamruzicka** [comment]
Would it make sense to introduce a bulk output endpoint for providing output for all hosts in the job matching a given search query while I'm at this?

**Comment by iNecas** [comment]
I think it would make sense to add, but probably a separate PR

**Comment by iNecas** [comment]
[test foreman]

**Comment by iNecas** [comment]
[test foreman_remote_execution]

**Comment by adamruzicka** [comment]
[test foreman_remote_execution]

File: app/controllers/api/v2/job_invocations_controller.rb [hunk 1]
```diff
@@ -177,6 +181,22 @@ def output_lines_since(task, time)
...
         line_sets
       end
+
+      def host_output(job_invocation, host, default: nil, since: nil, raw: false)
+        refresh = true
+
+        if (task = job_invocation.sub_task_for_host(host))
+          refresh = task.pending?
+          output  = output_lines_since(task, since)
+          output  = output.map { |set| set['output'] }.join("\n") if raw
```
[the remark this answers is not in the document]
Comment (adamruzicka) [reply app/controllers/api/v2/job_invocations_controller.rb]: If I drop the `if raw` part, then both the regular and raw endpoints behave the same way (the raw way).

[the remark this answers is not in the document]
Comment (iNecas) [reply app/controllers/api/v2/job_invocations_controller.rb]: Sry, I meant:

```ruby
output  = output.map { |set| set['output'] }.join if raw
```

**Comment by iNecas** [comment]
Test failures related

**Comment by iNecas** [comment]
Seems like jenkins still doesn't like something about it

**Comment by iNecas** [comment]
Works well now (tested with simulated proxy, as well as with the real one)

```
curl 2>/dev/null -k -u admin:changeme -H 'application/json' http://localhost:3000/api/v2/job_invocations/933/hosts/146/raw | jq -r .output
====== Simulated Remote Execution ======

This is an output of a simulated remote
execution run. It should run for about
5 seconds and finish successfully.
Exit status: 0
```

**Pull request merged**